### PR TITLE
feat: The HTTP module supports set Cookie

### PR DESCRIPTION
**Pull request by kkumar-gcc**
Closes https://github.com/goravel/goravel/issues/271

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: contracts/http/response.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ type Response interface {
 }
 
 type ContextResponse interface {
+	// Cookie adds a cookie to the response.
+	Cookie(cookie *Cookie) ContextResponse
```
Comment (hwbrzzl) [inline contracts/http/response.go:16]: S: Pointer is unnecessary.

```suggestion
	Cookie(cookie Cookie) ContextResponse
```

File: contracts/http/response.go [hunk 2]
```diff
@@ -37,6 +39,10 @@ type ContextResponse interface {
 	View() ResponseView
 	// Writer returns the underlying http.ResponseWriter associated with the response.
 	Writer() http.ResponseWriter
+	// WithCookie adds a cookie to the response.
+	WithCookie(cookie *Cookie) ContextResponse
```
Comment (hwbrzzl) [inline contracts/http/response.go:43]: S: Don't need this.

File: contracts/http/response.go [hunk 3]
```diff
@@ -37,6 +39,10 @@ type ContextResponse interface {
...
+	// WithoutCookie removes a cookie from the response by setting its expiration time to a past date.
```
Comment (hwbrzzl) [inline contracts/http/response.go:44]: Drivers can remove cookie by themselves, maybe not `setting its expiration time to a past date`

```suggestion
	// WithoutCookie removes a cookie from the response.
```

Comment (kkumar-gcc) [reply contracts/http/response.go]: Yeah, you are right.

File: contracts/http/response.go (hunk 2 above)
Comment (kkumar-gcc) [reply contracts/http/response.go]: Okay

File: contracts/http/response.go (hunk 1 above)
Comment (kkumar-gcc) [reply contracts/http/response.go]: Okay

File: contracts/http/cookie.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	Name     string
+	Value    string
+	Path     string
+	Domain   string
+	Expires  time.Time
+	MaxAge   int
+	Secure   bool
+	HttpOnly bool
+	Raw      string
+	SameSite string
```
[the remark this answers is not in the document]
Comment (kkumar-gcc) [reply contracts/http/cookie.go]: Yeah, sure.

**Review by hwbrzzl — APPROVED** [review]
Great!

**Pull request merged**